### PR TITLE
feat: Implementation-of-Smart-Lock-Transaction-Execution-Flow-112

### LINKED ISSUE #112: [CONTRACT] Implement Smart Lock Transaction Execution Flow
## Description 📝
Implement the functionality to execute approved smart lock transaction proposals in our multisignature account system. This feature will allow the system to automatically execute smart lock transactions once they have received the required number of approvals from authorized members.

The `execute_smart_token_lock_transaction` functionality will process approved smart lock proposals, interact with the treasury to lock the specified tokens, and update the transaction status accordingly. This builds upon the proposal system implemented in the `actions/smart_token_lock_transaction.cairo` component.

## Proposed Steps 📝
1. Implement the `execute_smart_token_lock_transaction` function that:
   - Takes a `transaction_id` as parameter
   - Validates that the transaction exists and is of type `SmartTokenLockTransaction`
   - Checks that the transaction has received sufficient approvals via the AccountData component's `is_transaction_approved` function
   - Validates that the transaction hasn't already been executed
   - Retrieves the `SmartTokenLockTransaction` details from storage

2. Integrate with TreasuryHandler component to execute the lock:
   - Call the `TreasuryHandler`'s token locking function with the transaction parameters
   - Pass the `token`, `amount`, and `duration` from the SmartTokenLockTransaction
   - Handle any potential failures during the lock execution

3. Update transaction status and storage:
   - Mark the transaction as executed in the AccountData component by calling the execute_transaction function

4. Implement proper error handling:
   - Validate transaction exists and is pending
   - Ensure sufficient token balance still exists at execution time
   - Handle cases where the lock operation fails
   - Provide meaningful error messages for different failure scenarios

5. Emit execution events:
   - Emit a `SmartTokenLockTransactionExecuted` event on successful execution
   - Include transaction_id, token, amount, duration, and execution timestamp
  

## Acceptance Criteria ✅
- Execution function properly validates transaction state and approvals
- Integration with TreasuryHandler component works correctly for token locking
- Transaction status is properly updated after execution
- Comprehensive error handling for all failure scenarios
- Appropriate events are emitted for both success and failure cases
- All edge cases are handled (insufficient balance, invalid transaction, etc.)
- All new logic is covered by unit tests
- Integration tests demonstrate end-to-end flow from proposal to execution

## Dependencies 📋
- Requires completion of the smart lock transaction proposal feature
- Depends on TreasuryHandler component having token locking capabilities
- Integrates with existing AccountData component for transaction management

### Notes:
- Follow the structure and conventions of existing transaction execution implementations
- Ensure seamless integration with the existing multisig approval workflow
- Document all new functions, events, and error conditions
- Maintain backward compatibility with existing transaction types
- Consider gas optimization for execution operations
- When you apply please refer to the TG group and drop a message stating interest on that particular issue
- Please star this repo to get assigned an issue. This helps increase our project's visibility.

Telegram: https://t.me/+qahDxioW8SA0NjRk

Blocked by #110 and #111

### PR DESCRIPTION
**Pull request by anihdev**
## Description 📝
This PR implements the `execute_smart_token_lock_transaction` functionality in the multisignature account system. This feature allows the system to automatically execute smart lock transactions once they have received the required number of approvals from authorized members.
I also implemented a comprehensive testing for the `execute_smart_token_lock_transaction` functionality which processes approved smart lock proposals, interacts with the treasury to lock the specified tokens, and updates the transaction status accordingly.

## Tests Added 
* `test_execute_smart_token_lock_transaction_success` - Tests successful execution.
* `test_execute_smart_token_lock_transaction_integration_flow` - Full integration test .
* `test_execute_smart_token_lock_transaction_not_approved` - Validates rejection of unapproved transactions.
* `test_execute_smart_token_lock_transaction_wrong_type` - Ensures proper validation of transaction types.
* `test_execute_smart_token_lock_transaction_insufficient_balance` - Tests insufficient token balance scenarios.
* `test_execute_smart_token_lock_transaction_already_executed` - Prevents double execution of transactions.
* `test_execute_smart_token_lock_transaction_when_paused` - Validates pause mechanism.

## Related Issues 🔗
Fixes #112 

## Changes Made 🚀
- [ ✅] ✨ Feature Implementation 

## Checklist ✅
- [✅] 🛠 I have tested these changes. 
- [✅] 📖 I have updated the documentation.
- [✅] 🎨 This PR follows the project's coding style. 
- [✅] 🧪 I have added necessary tests.

## Summary by CodeRabbit

* **New Features**
  * Added the ability to execute smart token lock transactions, including validation and event emission upon execution.
  * Introduced a new event to notify users when a smart token lock transaction has been executed.

* **Bug Fixes**
  * Improved error messaging for attempts to execute already completed transactions.

* **Tests**
  * Expanded test coverage to include scenarios for executing smart token lock transactions, covering both successful and failure cases.

### CODE REVIEW

File: src/account_data.cairo [hunk 1]
```diff
@@ -473,6 +473,11 @@ pub mod AccountData {
                 transaction.tx_status.read() == TransactionStatus::APPROVED,
                 Errors::ERR_TRANSACTION_NOT_EXECUTABLE
             );
+            // Ensure transaction hasn't been executed already
+            assert(
+                transaction.tx_status.read() != TransactionStatus::EXECUTED,
+                Errors::ERR_TRANSACTION_ALREADY_EXECUTED
+            );
```
Comment (codebestia) [inline src/account_data.cairo:480]: This won't matter because we checked already above that the tx_status is equal to APPROVED. Please remove this.

File: src/actions/smart_token_lock_transaction.cairo [hunk 2]
```diff
@@ -109,5 +126,58 @@ pub mod SmartTokenLockTransactionComponent {
...
+            );
+
+            // Validate transaction is approved and not executed
+            assert(
+                transaction.tx_status == TransactionStatus::APPROVED,
+                Errors::ERR_TRANSACTION_NOT_EXECUTABLE
+            );
+
+            // Smart token lock transaction details
+            let smart_lock_tx = self.smart_token_lock_transactions.entry(transaction_id).read();
```
Comment (codebestia) [inline src/actions/smart_token_lock_transaction.cairo:153]: This can all be done with `let transaction = self.get_smart_token_lock_transaction(transaction_id);`

**Review by codebestia — CHANGES_REQUESTED** [review]
Good Job @anihdev 
Just some minor fixes here and we are good.
